app/controllers/users_controller.rb

```diff
@@ -9,4 +16,55 @@ def show
     @user = User.find(params[:id])
     @posts = @user.posts.ordered_by_most_recent
   end
+
+  def friend?(user)
+    current_user.friends? user
+  end
+
+  def request_sent?(user)
+    user.friend_responses.one? { |u| u.id == current_user.id }
+  end
+
+  def request_received_from?(user)
+    current_user.friend_responses.detect { |response| response.id == user.id }
+  end
+
+  def send_friend_request
+    user_id = params[:user_id]
+    user2 = User.find user_id
+    Friendship.create(user1: current_user, user2: user2, status: 'Pending')
+    redirect_to user_path id: user_id
+  end
+
+  def send_friend_request_path(user)
+    "/send_friend_request?user_id=#{user.id}"
+  end
+
+  def accept_friend_request_path(user)
+    "/accept_friend_request?user_id=#{user.id}"
+  end
+
+  def reject_friend_request_path(user)
+    "/reject_friend_request?user_id=#{user.id}"
+  end
+
+  def friend_requests
+    @requests = current_user.pending_friend_requests
+  end
+
+  def friend_responses
+    @responses = current_user.friend_responses
+  end
+
+  def accept_friend_request
+    Friendship.where(user_1: params[:user_id], user_2: current_user.id)
+      .update(status: 'Accepted')
+    redirect_to '/friend_responses'
+  end
+
+  def reject_friend_request
+    Friendship.where(user_1: params[:user_id], user_2: current_user.id)
+      .update(status: 'Rejected')
+    redirect_to '/friend_responses'
+  end
```

Comment:
One of the most important ways to write clear and concise code in Ruby on Rails, the motto “Fat Model, Skinny Controller” refers to how the M and C parts of MVC ideally work together. Namely, any non-response-related logic should go in the model, ideally in a nice, testable method. Meanwhile, the “skinny” controller is simply a nice interface between the view and model. For more insights see this article on [10 Ruby on Rails Best Practices](https://www.sitepoint.com/10-ruby-on-rails-best-practices/)

You can move any logic that isn’t about the response (for example, setting a flash message, or choosing whether to redirect or render a view) to the model (instead of the controller),